### ISSUE TITLE
Where to find StartAllBackX64.dll compatible with the latest ExplorerPatcher?

### ISSUE THREAD

**Issue by jed84** [valinet/ExplorerPatcher#44]
I am having problems to find the latest version of StartAllBackX64.dll with SHA-256: D8D1484D6F7DE94C5018B03A5A93EE8BBB7795BCA5408A721CBB74BE28065689)
In what version of StartAllBack is it included? It doesn't look like it's from 2.9.91 RC.

I think it would be helpful to link to the version compatible with ExplorerPatcher to prevent breaking the OS.

**Comment by gabri25ele** [comment]
so we need to install startisback, together with each other?

**Comment by valinet** [comment]
@jed84 I don’t have a license from StartAllBack’s developer to redistribute their files, that’s why you have to find the DLL on your own. Only a small subset of features require that DLL, nothing deal breaking regarding the core functionality of the patcher, so that’s why I don’t also update for the latest release of SB as soon as it is released. It is not a priority. Ideally, all features used from SB would be ported over, but I can’t do that and all the other stuff all alone instantly, and this Github seems useful only to attract tons of questions and requests and almost no one interested to get involved. Anyway, again, you like to overstate things as others around here have also done: “prevent breaking the OS” - the README clearly states what SHA-256 the DLL has to have. If you can’t find that and use another version thinking it will work, it won’t. It is a computer program, it doesn’t have intelligence, it doesn’t know of human concepts like hope that it works or tricks like I’ll use the DLL from the latest version, it just does what it is programmed to do: it looks for stuff in SB’s DLL at certain well defined places, and the stuff is only there on a certain version, the one I referred to. Any other version has a completely changed internal layout and it won’t work. It’s simple. These DLLs are not generated manually, but by a compiler and a linker. These 2 guys are not mandated to place stuff at the same locations as they did in the previous version, for a variety of reasons including optimization of the produced binary, so they don’t do it, hence why the new version differs completely from an internal layout perspective compared to the right version, even though few actual code changes may have happened between the 2. Again, if one follows the instructions, nothing breaks.

> so we need to install startisback, together with each other?
> can't i just take only the dll and use only that one?
> for example the new dll is that
> [StartAllBackX64.zip](https://github.com/valinet/ExplorerPatcher/files/7315267/StartAllBackX64.zip)

I don’t know where in the README it says to install StartAllBack. It says that for a certain limited subset of features, you can provide the application with a copy of the StartAllBack DLL placed in a certain location. It says nothing about actually have StartAllBack installed. The StartAllBack installer is also an archive. You can easily open the exe they provide with 7zip and extract the DLL from there. Again, as explained above, I don’t have a license to redistribute their files, so it’s the interested users’ responsibility to take care of this aspect.

**Comment by gabri25ele** [comment]
> 
> 
> Only 

answering with such acidity doesn't help anyone

**Comment by jed84** [comment]
Thank you, the file comes from v.2.9.90

**Comment by valinet** [comment]
> > Only
> 
> answering with such acidity doesn't help anyone

Yeah, I understand, but try to understand the other party as well. I receive a lot of these messages daily, to the point that I barely have time to work on the actual stuff. I am not a multi billion dollar corporation with separate support people, separate developers etc. I am a single person in charge of everything, coming with the human things as well, like getting pissed off sometimes etc., for better or worse. Most of the “issues” here are repetitive and not really issues regarding the project. Despite having made separate threads for people to post into, to keep everything organized and in the hope that other users would help as well, no one wants to comply with that. It is just easier to post a new issue, why take the time to browse through old threads, why take the time to actually follow some directions. And it’s not only this thread, almost all threads follow this pattern. The issues panel is designed so the developers can track the real problems. I can’t track real problems if it is filled up with support questions. I made a separate thread for that, almost on one is interested, so, how should I feel, happy and non-acidic? Have you seen the other open projects here on GitHub? A lot of developers just give up on the issues panel and ignore it completely, as it just means wasting all the time scrubbing through repetitive stuff, which many frankly do not have the time to. They just give up…

As I said, many people here propose stuff. No one follows up with a pull request. It’s easier just to not bother and throw it on the developer’s desk. And I understand, not everyone’s a programmer, but a lot of stuff propose changes to the README, for example. For that, anyone can write a pull request. This project is made in the free time of people, including me, it gets better the more the people get involved, “sacrificing“ some of their free time to contribute something back. It would be awesome and I’d really appreciate a lot if people would get actually hands on involved, even for the small things, like patching a README, everything counts.

It is easy to feel like I am acidic for no reason, but think about the way other projects tackle support and suggestions: some just develop behind close doors, they put up new versions and people discuss and help each other on a forum somewhere not associated with the developer, where answers to questions come mainly from other parties involved. Others just don’t even bother with users suggestions at all and do whatever they feel like doing. And others just charge money for support. Yet you (and probably others as well) get mad at some guy doing participative development, where people can see what it is currently being worked on and make suggestions, that always answers the (same repetitive) questions and who is doing all of this in his free time for free… yeah, seems fair.

I am starting to believe less and less in this “developing in the open” model. I still strongly believe in the value proposition of having access to source code for learning purposes, at least, but that could be achieved via other ways. But to say that I am disappointed in the actual community involvement in this project, beside questions, suggestions and the few people that bother to help others after they got their answer, is an understatement. And it’s happened multiple times, already, to the point that I don’t know why I still bother… also, what’s the point of doing all of this if in the end no one else contributes to this project, reviews files etc? I could just release periodic builds and the source code, and just develop behind closed doors. GitHub was meant as a developer’s platform, yet all I see is a lot of support tickets… and the funny thing is someone warned me about this a year ago, but I kind of dismissed it, not fully understood him at the time. Now I do. Oh, the irony. I guess some things you understand only when you are personally faced with.

Also, how many times have you or others complained to Microsoft for the recent changes in Windows that few like? How have they responded, did they even bother to reply something other than the standard templates they use? It is not my or anyone else’s responsibility for fixing their shit. I admit, my replies may have been on the acidic side, but sure it’s easy to get mad at me, that I actually communicate, compared to Microsoft, for example, whose stuff is broken and who do not even bother to engage in a conversation with their users, so of course they can’t be acidic.

Anyway, I am sorry for sounding harsh or acidic as you said, which was the right word btw, towards you. It is not aimed or targeted towards you specifically, it is just a general feeling that I developed more and more recently and I happened to expose it in this thread, nothing more. I don’t have anything against you or anyone participating here, and certainly would not like you or anyone else feeling that way. I just have a few issues of my own with how stuff goes around here which I have explained above mostly. Again, I deeply apologize for sounding acidic towards you and hope that you and the others interested have bothered to read the entirety of this post and have understood the reasons exposed.

Thank you.
